omod-dstu3/src/main/java/org/openmrs/module/fhir2/providers/EncounterDstu3FhirResourceProvider.java

```diff
@@ -0,0 +1,82 @@
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.providers;
+
+import javax.validation.constraints.NotNull;
+
+import java.util.List;
+
+import ca.uhn.fhir.rest.annotation.*;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.convertors.VersionConvertor_30_40;
+import org.hl7.fhir.dstu3.model.*;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.openmrs.module.fhir2.api.FhirEncounterService;
+import org.openmrs.module.fhir2.util.FhirServerUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Qualifier("fhirDstu3Resources")
+@Setter(AccessLevel.PACKAGE)
+public class EncounterDstu3FhirResourceProvider implements IResourceProvider {
+	
+	@Autowired
+	private FhirEncounterService encounterService;
+	
+	@Override
+	public Class<? extends IBaseResource> getResourceType() {
+		return Encounter.class;
+	}
+	
+	@Read
+	@SuppressWarnings("unused")
+	public Encounter getEncounterById(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.Encounter encounter = encounterService.getEncounterByUuid(id.getIdPart());
+		if (encounter == null) {
+			throw new ResourceNotFoundException("Could not find encounter with Id " + id.getIdPart());
+		}
+		
+		return VersionConvertor_30_40.convertEncounter(encounter);
+	}
+	
+	@Search
+	public Bundle searchEncounter(@OptionalParam(name = Encounter.SP_DATE) DateRangeParam date,
+	        @OptionalParam(name = Encounter.SP_LOCATION, chainWhitelist = { "", Location.SP_ADDRESS_CITY,
+	                Location.SP_ADDRESS_STATE, Location.SP_ADDRESS_COUNTRY,
+	                Location.SP_ADDRESS_POSTALCODE }, targetTypes = Location.class) ReferenceAndListParam location,
+	        @OptionalParam(name = Encounter.SP_PARTICIPANT, chainWhitelist = { "", Practitioner.SP_IDENTIFIER,
+	                Practitioner.SP_GIVEN, Practitioner.SP_FAMILY,
+	                Practitioner.SP_NAME }, targetTypes = Practitioner.class) ReferenceAndListParam participantReference,
+	        @OptionalParam(name = Encounter.SP_SUBJECT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_GIVEN,
+	                Patient.SP_FAMILY,
+	                Patient.SP_NAME }, targetTypes = Patient.class) ReferenceAndListParam subjectReference) {
+		return VersionConvertor_30_40.convertBundle(FhirServerUtils.convertSearchResultsToBundle(
+		    encounterService.searchForEncounters(date, location, participantReference, subjectReference)));
+		
+	}
+	
+	@History
+	@SuppressWarnings("unused")
+	public List<Resource> getEncounterHistoryById(@IdParam @NotNull IdType id) {
```

Comment:
I'd prefer if these files were structured similarly, i.e. with the `@History` method following the `@Read` method.

Reply:
@ibacher This might need to be changed in master as well. I am not changing anything in omod  module because it might be irrelevant for this PR